src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Common/TensorPrimitives.IBooleanUnaryOperator.cs

```diff
@@ -0,0 +1,751 @@
+            }
+
+            return TAnyAll.DefaultResult;
+        }
+
+        /// <summary>Performs an element-wise operation on <paramref name="x"/> and writes the results to <paramref name="destination"/>.</summary>
+        /// <typeparam name="T">The element input type.</typeparam>
+        /// <typeparam name="TOperator">Specifies the operation to perform on each element loaded from <paramref name="x"/>.</typeparam>
+        private static void InvokeSpanIntoSpan<T, TOperator>(
+            ReadOnlySpan<T> x, Span<bool> destination)
+            where TOperator : struct, IBooleanUnaryOperator<T>
+        {
+            if (x.Length > destination.Length)
+            {
+                ThrowHelper.ThrowArgument_DestinationTooShort();
+            }
+
+            if (sizeof(T) == 1)
+            {
+                Vectorized_Size1(x, destination);
+            }
+            else if (sizeof(T) == 2)
+            {
+                Vectorized_Size2(x, destination);
+            }
+            else if (sizeof(T) == 4)
+            {
+                Vectorized_Size4(x, destination);
+            }
+            else
+            {
+                Vectorized_Size8OrOther(x, destination);
+            }
+
+            static void Vectorized_Size1(ReadOnlySpan<T> x, Span<bool> destination)
+            {
+                Debug.Assert(sizeof(T) == sizeof(bool));
+
+                ref T xRef = ref MemoryMarshal.GetReference(x);
+                ref bool destinationRef = ref MemoryMarshal.GetReference(destination);
+                int i = 0;
+
+                if (Vector512.IsHardwareAccelerated && TOperator.Vectorizable && Vector512<T>.IsSupported)
+                {
+                    int vectorFromEnd = x.Length - Vector512<T>.Count;
+                    if (i <= vectorFromEnd)
+                    {
+                        // Loop handling one input vector / one output vector at a time.
+                        do
+                        {
+                            Process(ref xRef, ref destinationRef, i);
+                            i += Vector512<T>.Count;
+                        }
+                        while (i <= vectorFromEnd);
+
+                        // Handle any remaining elements with a final vector.
+                        if (i != x.Length)
+                        {
+                            i = x.Length - Vector512<T>.Count;
+                            Process(ref xRef, ref destinationRef, i);
+                        }
+
+                        return;
+
+                        static void Process(ref T xRef, ref bool destinationRef, int i)
+                        {
+                            Vector512<byte> v = TOperator.Invoke(Vector512.LoadUnsafe(ref xRef, (uint)i)).AsByte();
+
+                            Vector512.ConditionalSelect(v, Vector512<byte>.One, Vector512<byte>.Zero)
+                                .StoreUnsafe(ref Unsafe.As<bool, byte>(ref destinationRef), (uint)i);
```

Comment:
Are all the usages cases where `v` is known to be `AllBitsSet` vs `Zero` on a per-element basis?

Can we simplify this to `v & V512<byte>.One` instead?

Reply:
Probably. I'll double check and change it if so